pycqed/measurement/waveform_control/block.py

```diff
@@ -94,8 +94,11 @@ def build(self, ref_point="end", ref_point_new="start",
 
             # rename ref pulse within the block if not a special name
             escape_names = ("previous_pulse", "segment_start")
-            if ref_pulse not in escape_names and not p_is_block_start:
-                p['ref_pulse'] = name + "-|-" + p['ref_pulse']
+            if isinstance(ref_pulse, list):
+                p['ref_pulse'] = [name + "-|-" + rp for rp in p['ref_pulse']]
```

Comment:
Here if I understand correctly you assume that in the list there will never be a name that shouldn't be renamed. Do I understand this correctly? Is this what we want? We could maybe still include the  check in the inline for loop, i.e. change line 98 to:  p['ref_pulse'] = [name + "-|-" + rp for rp in p['ref_pulse'] if rp not in escape_names and not p_is_block_start]

Reply:
The reason that I do not have this here is that the multiref code in segment.py is not implemented for these aliase names. If we want the multiref feature to also work for these alias names, further thoughts and further code changes in segment.py would be needed. I would suggest to add this to the list of issues to be solved in the future. Justification: The aliases are only for convenience. Any timing that could be created with these aliases could for now also be created by directly referencing pulse names. As soon as the segment code can handle it, you proposed change here of course would make sense.

Reply:
mmmh interesting, I see. Then I guess that a simultaneous block cannot be used within another block. Is that correct? because in that case, the higher level block would add the aliases to all pulse names also in the lower level (compiled) block and therefore the multiref in segment.py would fail to recognize the pulses. Do I get this right or not? If so I'm fine to have that as a limitation for now but then I would add a comment in the docstring in both docstring of block.build() and simultaneous_blocks() just so that the user can see it. Would that make sense? something like: "When using multi-reference pulses, the resulting block cannot be included in a higher order level block".
We should indeed add this as an issue and think of a better way to handle those aliases.

Reply:
No, there is no problem if other pulses use these alias names. Only for pulses, for which a list is given as ref_pulse, the list should contain only actual pulse names. I have updated the docstring of UnresolvedPulse to clarify this.

Further info about the issue: segment_start is treated completely separate in resolve_timing before any other timing is resolved. Thus, a change in the structure of resolve_timing would be needed to allow multiple references where one of them is segment_start. I think I know how to do this efficiently and could implement it as soon as someone volunteers for rigorous testing. As a workaround, you could for now add a virtual pulse referenced to segment_start, and then use this pulse in your list. There should be no obstacle for extending the multiref code to allow previous_pulse, but there are situations in which segment replaces previous_pulse by segment_start, so the segment_start issue should be solved first.